Clustered RabbitMQ sample (#5321)
Particular/docs.particular.net: samples/rabbitmq/cluster/sample.md

```

## Code walk-through

This sample shows how to use a clustered RabbitMQ as a transport for NServiceBus to connect two endpoints. The sender either sends a command or publishes an event to a receiver endpoint, via the RabbitMQ broker, and writes to the console when the message is received.

### Configuration

snippet: cluster-configuration

The username and password can be configured via the connection string. If these are not present, the connection string defaults to `host=localhost;username=guest;password=guest`.

Note that [delayed delivery](/nservicebus/messaging/delayed-delivery.md) isn't supported for clusters and the sample therefor disables them. This means that [saga timeouts](/nservicebus/sagas/timeouts) can't be used and [delayed retries](https://docs.particular.net/nservicebus/recoverability.md#delayed-retries) must be disabled as shown below:

snippet: cluster-disable-retries

### Adding nodes

RabbitMQ allows additional nodes to be added to avoid having to use a load balancer in front of the cluster. Adding those nodes will make the connection manager use them in a round robin fashion and automatically re-connect to healthy nodes should the connection be lost.

> Since RabbitMQ will default to port `5672` we only have to add nodes `rabbit2` and `rabbit3` as shown below:

Reviewer: ```suggestion
RabbitMQ will default to port `5672` therefore the extra nodes (`rabbit2` and `rabbit3`) are added as shown below:
```
